seaborn/tests/test_axisgrid.py

```diff
@@ -781,6 +781,29 @@ def test_map_diag(self):
             nt.assert_equal(len(ax.patches), 40)
 
     @skipif(old_matplotlib)
+    def test_map_diag_color(self):
+        color_set = {mpl.colors.cnames[x].lower()
+                     for x in ['red', 'white', 'black']}
+
+        g1 = ag.PairGrid(self.df)
+        g1.map_diag(plt.hist, color='red')
+
+        for ax in g1.diag_axes:
+            colors = [mpl.colors.rgb2hex(patch.get_facecolor()).lower()
+                      for patch in ax.patches]
+            for color in colors:
+                nt.assert_true(color in color_set, color)
```

Comment:
Well I think that both of your questions are tied together; I must have tested for the wrong thing, and the patches have the line (which should be red) and other objects (which are black and white).
If you can suggest a better test for this let me know, I don't have experience testing plotting libraries.